timer_rate: 50.0
invalidate_time: 2.0 
